common/src/main/java/io/netty5/util/concurrent/AbstractTaskScheduler.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2023 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.netty5.util.concurrent;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+
+import static io.netty5.util.concurrent.AbstractScheduledEventExecutor.deadlineNanos;
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.Executors.callable;
+
+/**
+ * Abstract base class for {@link TaskScheduler} that want to support scheduling.
+ *
+ * @see AbstractScheduledEventExecutor
+ */
+public abstract class AbstractTaskScheduler implements TaskScheduler {
+
+    /**
+     * Returns a {@link RunnableScheduledFuture} for the given values.
+     */
+    protected abstract  <V> RunnableScheduledFuture<V> newScheduledTaskFor(Callable<V> callable,
+                                                                           long deadlineNanos, long period);
+
+    protected abstract Ticker ticker();
+
+    @Override
+    public Future<Void> schedule(Runnable command, long delay, TimeUnit unit) {
+        requireNonNull(command, "command");
+        requireNonNull(unit, "unit");
+        if (delay < 0) {
+            delay = 0;
+        }
+        RunnableScheduledFuture<Void> task = newScheduledTaskFor(
+                callable(command, null), deadlineNanos(ticker().nanoTime(), unit.toNanos(delay)), 0);
+        return schedule(task);
```

Comment:
**Scalability Issue**: The schedule method in AbstractTaskScheduler directly creates a new instance of RunnableScheduledFuture for each scheduled task. This could lead to a high rate of object creation and garbage collection in systems with a high volume of task scheduling operations, impacting scalability and performance. <br> **Fix**: Utilize a pool of RunnableScheduledFuture objects instead of instantiating new ones for each task. This object pooling pattern can significantly reduce the rate of object creation and garbage collection, improving the system's scalability and performance. <br> **Code Suggestion**: 
 ```
 +    private final ObjectPool<RunnableScheduledFuture<?>> futurePool;
 +    public AbstractTaskScheduler(ObjectPool<RunnableScheduledFuture<?>> futurePool) {
 +        this.futurePool = futurePool;
 +    }

 +    @Override
 +    public Future<Void> schedule(Runnable command, long delay, TimeUnit unit) {
 +        RunnableScheduledFuture<Void> task = futurePool.get();
 +        task.reset(command, delay, unit);
 +        return schedule(task);
 +    }
 ```